docs(decisions): add Architecture Decision Record (ADR) framework (#19767)
carbon-design-system/carbon: docs/decisions/0001-record-project-decisions.md

# 1. Record project decisions

Date: 2025-06-24

## Status

Accepted

## Context

As the Carbon Design System has evolved over time, the context and reasoning
surrounding many decisions has been lost.

We need to record the high level decisions made on this project to preserve this

> chain of history both for ourselves as well as future maintainers.

Reviewer: ```suggestion
chain of history for ourselves and future maintainers.
```
